packages/trader/src/AppV2/Containers/ContractDetails/contract-details.tsx

```diff
@@ -0,0 +1,110 @@
+import ContractDetailsFooter from 'AppV2/Components/ContractDetailsFooter';
+import { ContractCard } from 'AppV2/Components/ContractCard';
+
+const ContractDetails = observer(() => {
+    const { contract_info, is_loading } = useContractDetails();
+    const { contract_id, currency, contract_type, limit_order } = contract_info;
+    const [update_history, setUpdateHistory] = React.useState<TContractUpdateHistory>([]);
+    const { common } = useStore();
+    const { server_time } = common;
+    const { is_take_profit_visible, is_stop_loss_visible } = getContractDetailsConfig(contract_type ?? '');
+    const is_valid_to_sell = isValidToSell(contract_info);
+    type TContractUpdateHistory = TContractStore['contract_update_history'];
+    type TResponse = {
+        contract_update_history: TContractUpdateHistory;
+    };
+
+    const getSortedUpdateHistory = (history: TContractUpdateHistory) =>
+        history.sort((a, b) => Number(b?.order_date) - Number(a?.order_date));
+    const requestUpdatedHistory = React.useCallback((id?: number) => {
+        if (!id) return;
+        WS.contractUpdateHistory(id)
+            .then((response: TResponse) => {
+                setUpdateHistory(getSortedUpdateHistory(response.contract_update_history));
+            })
+            .catch(() => null);
+    }, []);
+
+    React.useEffect(() => {
+        requestUpdatedHistory(contract_id);
+    }, [contract_id, limit_order, requestUpdatedHistory]);
+
+    if (is_loading) return <></>;
+
+    const is_multiplier = isMultiplierContract(contract_info.contract_type);
+
+    const is_valid_to_cancel = isValidToCancel(contract_info);
+    const should_show_sell =
+        (hasContractEntered(contract_info) ||
+            isForwardStarting(contract_info?.shortcode ?? '', contract_info.purchase_time)) &&
+        isOpen(contract_info);
+    const { is_tp_history_visible, is_deal_cancellation_visible } = getContractDetailsConfig(
+        contract_info.contract_type ?? ''
+    );
+    const show_cancel_button = is_multiplier && is_valid_to_cancel;
+    let showTpSl =
+        isOpen(contract_info) &&
+        (is_take_profit_visible || is_stop_loss_visible) &&
+        (is_valid_to_sell || is_deal_cancellation_visible);
+
+    if (isAccumulatorContract(contract_info.contract_type)) {
+        showTpSl = isOpen(contract_info) && Boolean(limit_order);
+    }
+    return (
+        <div
+            className={classNames('contract-details', {
+                'contract-details--two-buttons': should_show_sell && show_cancel_button,
+                'contract-details--one-button': should_show_sell && !show_cancel_button,
+            })}
+        >
+            <div className='contract-card-wrapper'>
+                <ContractCard contractInfo={contract_info} serverTime={server_time} />
+            </div>
+            <div className='placeholder'>
+                <ChartPlaceholder />
+            </div>
+            <DealCancellation />
+            {showTpSl && (
```

Comment:
`showTpSl` ---> `showRiskManagement`
Maybe it'll be better?

Reply:
sure